plugin/code_lens.py

```diff
@@ -131,10 +133,36 @@ def resolved_code_lens(self) -> Iterable[CodeLensData]:
             if lens.is_resolved():
                 yield lens
 
-    def render(self) -> None:
-        accent = self.view.style_for_scope("region.greenish markup.accent.codelens.lsp")["foreground"]
-        for index, lens in enumerate(self._flat_iteration()):
-            self.view.add_regions(self._region_key(index), [lens.region], "", "", 0, [lens.small_html], accent)
+    def _get_phantom_region(self, region: sublime.Region) -> sublime.Region:
+        """Takes the current region and puts the phantom on the line above, properly indented.
+
+        Code-lenses are typically shown above the code in VSCode.
+        """
+        line = self.view.substr(self.view.line(region))
+        offset = 0
+        for ch in line:
+            if ch.isspace():
+                offset += 1
+            else:
+                break
+
+        row, _ = self.view.rowcol_utf16(region.a)
+        point = self.view.text_point_utf16(max(row - 1, 0), offset, clamp_column=True)
+        return sublime.Region(point)
+
+    def render(self, mode: str) -> None:
+        if mode == 'annotation':
+            accent = self.view.style_for_scope("region.greenish markup.accent.codelens.lsp")["foreground"]
+            for index, lens in enumerate(self._flat_iteration()):
+                self.view.add_regions(self._region_key(index), [lens.region], "", "", 0, [lens.small_html], accent)
+        else:  # 'phantom'
```

Comment:
Since the `annotation` is the default option, it would be better to have it in the `else` branch so that we 
default to that when the setting is set to an unknown value.